pkg/rc/replication_controller.go

```diff
@@ -162,6 +165,7 @@ func New(
 	logger logging.Logger,
 	alerter alerting.Alerter,
 	healthChecker checker.ConsulHealthChecker,
+	artifactRegistry artifact.Registry,
```

Comment:
you need to pass this `artifactRegistry` into the `&replicationController{}` that this function returns